src/connectors/autocomplete/__tests__/connectAutocomplete-test.ts

```diff
@@ -0,0 +1,421 @@
+  createDisposeOptions,
+} from '../../../../test/mock/createWidget';
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import { createSingleSearchResponse } from '../../../../test/mock/createAPIResponse';
+import connectAutocomplete from '../connectAutocomplete';
+import { TAG_PLACEHOLDER } from '../../../lib/escape-highlight';
+
+describe('connectAutocomplete', () => {
+  it('throws without render function', () => {
+    expect(() => {
+      // @ts-ignore
+      connectAutocomplete();
+    }).toThrowErrorMatchingInlineSnapshot(`
+"The render function is not valid (got type \\"undefined\\").
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/autocomplete/js/#connector"
+`);
+  });
+
+  it('warns when using the outdated `indices` option', () => {
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+
+    const trigger = () => {
+      // @ts-ignore outdated `indices` option
+      makeWidget({ indices: [{ label: 'foo', value: 'foo' }] });
+    };
+
+    expect(trigger).toWarnDev(
+      '[InstantSearch.js]: Since InstantSearch.js 4, `connectAutocomplete` infers the indices from the tree of widgets.\nThe `indices` option is ignored.'
+    );
+  });
+
+  it('is a widget', () => {
+    const render = jest.fn();
+    const unmount = jest.fn();
+
+    const customAutocomplete = connectAutocomplete(render, unmount);
+    const widget = customAutocomplete({});
+
+    expect(widget).toEqual(
+      expect.objectContaining({
+        $$type: 'ais.autocomplete',
+        init: expect.any(Function),
+        render: expect.any(Function),
+        dispose: expect.any(Function),
+        getConfiguration: expect.any(Function),
+      })
+    );
+  });
+
+  it('renders during init and render', () => {
+    const searchClient = createSearchClient();
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+    const widget = makeWidget({});
+
+    expect(renderFn).toHaveBeenCalledTimes(0);
+
+    const helper = algoliasearchHelper(searchClient, '', {});
+    helper.search = jest.fn();
+
+    widget.init!(createInitOptions({ helper }));
+
+    expect(renderFn).toHaveBeenCalledTimes(1);
+    expect(renderFn).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        currentRefinement: '',
+        indices: [],
+        refine: expect.any(Function),
+      }),
+      true
+    );
+
+    widget.render!(createRenderOptions());
+
+    expect(renderFn).toHaveBeenCalledTimes(2);
+    expect(renderFn).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        currentRefinement: '',
+        indices: expect.any(Array),
+        refine: expect.any(Function),
+      }),
+      false
+    );
+  });
+
+  it('consumes the correct indices', () => {
+    const searchClient = createSearchClient();
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+    const widget = makeWidget({ escapeHTML: false });
```

Comment:
```suggestion
    const widget = makeWidget();
```

It doesn't make sense to add the option for test, we don't use it.

Reply:
This is needed otherwise we cannot do this assertion:

```ts
expect(secondRenderOptions.indices[0].hits).toEqual(firstIndexHits);
```

Because we add `__escaped`, to the array with `escapeHTML` to `true`. Jest considers them as different arrays.

Reply:
My bad.